plugins/inputs/wireless/wireless_mac.go

```diff
@@ -0,0 +1,56 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+)
+
+// default executable path & flags
+const (
+	OSXCMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport"
+	FLAG   = "-I"
+)
+
+func (w *Wireless) Gather(acc telegraf.Accumulator) error {
+	cmd := exec.Command(OSXCMD, FLAG)
+	wireless, err := internal.StdOutputTimeout(cmd, 2*time.Second) // cmd.Output()
+	if err != nil {
+		return err
+	}
+	metrics, tags := w.loadMacWirelessTable(wireless)
+	acc.AddFields("wireless", metrics, tags)
+	return nil
+}
+
+func (w *Wireless) loadMacWirelessTable(table []byte) (map[string]interface{}, map[string]string) {
+	lines := strings.Split(string(table), "\n")
+	tags := make(map[string]string)
+	fields := make(map[string]interface{})
+	for _, line := range lines {
+		fm := strings.SplitN(strings.TrimSpace(line), ":", 2)
+		if len(fm) < 2 || fm[1] == "" {
+			continue
+		}
+		name := strings.Replace(strings.Trim(strings.TrimSpace(fm[0]), ":"), " ", "_", -1)
+		v := strings.TrimSpace(fm[1])
+		val, err := strconv.Atoi(v)
+		if err == nil { // it's a number
+			fields[name] = int64(val)
+		} else { // it's a string
+			if name == "channel" || name == "BSSID" || name == "SSID" {
+				fields[name] = strings.Replace(v, " ", "_", -1)
+			} else {
+				tags[name] = strings.Replace(v, " ", "_", -1)
+			}
+		}
```

Comment:
How about
```suggestion
		v := strings.TrimSpace(fm[1])
		switch name {
		case "channel", "BSSID", "SSID":
			fields[name] = strings.Replace(v, " ", "_", -1)
		default:
			if val, err := strconv.Atoi(v); err == nil {
				// it's a number
				fields[name] = int64(val)
			} else {
				// it's a string
				tags[name] = strings.Replace(v, " ", "_", -1)
			}
		}
```
This would allow to easily add more "special handlings" for certain properties if needed.

Reply:
¯\_(ツ)_/¯ Ok I guess.

Reply:
As I wrote, it's not a must, but I don't like these "type-guessings" and rather prefer to explicitly state expectations for certain fields in the code. Especially in cases where the output of tools changes this has proven valueable...

Reply:
I've made the change. Understand the reasoning. Thanks. I'll submit the change once the above open discussion is finalized.